packages/calypso-e2e/src/lib/pages/login-page.ts

```diff
@@ -43,75 +54,113 @@ export class LoginPage {
 	private async pageSettled(): Promise< void > {
 		// Needs to be `networkidle`, otherwise switching accounts will fail.
 		await this.page.waitForLoadState( 'networkidle' );
-		const container = await this.page.waitForSelector( selectors.loginContainer );
-		// The login container can fade in or shift, so wait for that to complete.
-		await container.waitForElementState( 'stable' );
+	}
+
+	/**
+	 * Navigates to the /log-in endpoint.
+	 */
+	async visit(): Promise< void > {
+		await this.page.goto( getCalypsoURL( 'log-in' ), { waitUntil: 'networkidle' } );
 	}
 
 	/**
 	 * Switch account if an user is already logged in.
 	 */
 	private async switchAccount(): Promise< void > {
-		// By default, log out of the existing account (even if test steps end up logging back in).
-		const alreadyLoggedIn = await this.page.$( selectors.changeAccountButton );
+		await this.pageSettled();
+
+		// Change account button takes some time to fade into view if an account
+		// is already logged in.
+		const alreadyLoggedIn = await this.page.isVisible( selectors.continueAsUser );
 		if ( alreadyLoggedIn ) {
-			console.log( 'already logged in, selecting "change account' );
+			const elementHandle = await this.page.waitForSelector( selectors.continueAsUser );
+			await elementHandle.waitForElementState( 'stable' );
 			await this.page.click( selectors.changeAccountButton );
 		}
 	}
 
 	/**
-	 * Executes series of interactions on the log-in page to log in as a specific user.
+	 * The basic flow of filling out username/email and password into the form then submitting.
 	 *
-	 * @param {Object} param0 Key/value pair holding the credentials for a user.
-	 * @param {string} param0.username Username of the user.
-	 * @param {string} param0.password Password of the user.
-	 * @returns {Promise<void>} No return value.
+	 * Call this method from any page to log into WordPress.com.
+	 *
+	 * @param param0 LoginCredentials object.
+	 * @param {string} param0.username User name.
+	 * @param {string} param0.password Password.
 	 */
-	async login( { username, password }: { username: string; password: string } ): Promise< void > {
-		await this.pageSettled();
-
-		await this.switchAccount();
-
-		// Begin the process of logging in.
+	private async baseflow( { username, password }: LoginCredentials ): Promise< void > {
```

Comment:
Note the baseflow (which interact with the actual Login page) only accepts the username/email and password pair.